# Spins, Magnetism, and Spin-Orbit Coupling

## EX.1 First Contact with Spins



### 🔎 1. What is Electron Spin?

Electron spin is a fundamental quantum property:

- Quantum number: $( s = \frac{1}{2} )$
- Two possible states: **spin-up (↑)** and **spin-down (↓)**
- Spin gives rise to an intrinsic **magnetic moment** for each electron.

Magnetic moment unit:

$[
mu_B = \frac{e\hbar}{2m_e}
]$

where:
- $(e)$ is electron charge
- $(\hbar)$ is reduced Planck's constant
- $(m_e)$ is electron mass

---

###  2. Why Magnetism Appears in Materials

Materials exhibit magnetism when there are **unpaired electrons**:

$[
N_\uparrow \neq N_\downarrow
]$

Here:
- $(N_\uparrow)$: number of spin-up electrons
- $(N_\downarrow)$: number of spin-down electrons

The **net magnetic moment** per unit cell:

$[
M = N_\uparrow - N_\downarrow
]$

This value is reported in Bohr magnetons (μB).

---

### 3. Spin in Standard (Non-Magnetic) DFT

In non-spin DFT:

- All electrons are treated as paired:
  $[
  n_\uparrow(\mathbf r) = n_\downarrow(\mathbf r)
  ]$
- Only total charge density is considered:
  $[
  n(\mathbf r) = n_\uparrow + n_\downarrow
  ]$
- Magnetism **cannot be described**.

The result therefore does *not* show magnetic effects even if the real material is magnetic.

---

###  4. Spin-Polarized DFT — The Key Concept

Spin-polarized DFT treats spin explicitly by considering:

$[
n_\uparrow(\mathbf r) \quad \text{and} \quad n_\downarrow(\mathbf r)
]$

which yields:

- **Total charge density**:
  $[
  n(\mathbf r) = n_\uparrow + n_\downarrow
  ]$

- **Spin density**:
  $[
  m(\mathbf r) = n_\uparrow - n_\downarrow
  ]$

These quantities are used to compute the effective potential for each spin channel.

The spin-dependent Kohn–Sham equations become:

$[
H_\uparrow \psi_\uparrow = \epsilon_\uparrow \psi_\uparrow
]$
$[
H_\downarrow \psi_\downarrow = \epsilon_\downarrow \psi_\downarrow
]$

This allows the system to find a spin-polarized ground state if energetically favorable.

---

###  5. Why Including Spin Lowers Energy

Two key effects play a role:

1. **Pauli Exclusion Principle**  
   Electrons of same spin avoid each other spatially, reducing Coulomb repulsion.

2. **Exchange Interaction**  
   This quantum effect stabilizes parallel spins; it lowers total energy when spins are aligned.

Physically, this is why materials like Fe, Co, and Ni are ferromagnetic — their total energy is lower in a spin-polarized state.

---




# make a new directory named EX1 inside the 01_Fe_bcc and copied all input file i.e fe_bcc.fdf and Fe.psf to EX1

In [1]:
!pwd

/home/l-rishiraj/siesta-docs/work-files/tutorials/basic/magnetism/01_Fe_bcc


In [2]:
ls

fe_bcc.fdf  Fe.psf  Untitled.ipynb


In [3]:
!mkdir -p EX1

In [4]:
!cp fe_bcc.fdf  Fe.psf EX1/

In [20]:
%cd EX1

/home/l-rishiraj/siesta-docs/work-files/tutorials/basic/magnetism/01_Fe_bcc/EX1


In [6]:
ls

fe_bcc.fdf  Fe.psf


In [9]:
!cat fe_bcc.fdf

#General system specifications
SystemName          Iron (bcc)
SystemLabel         fe_bcc
NumberOfAtoms       1
NumberOfSpecies     1

%block ChemicalSpeciesLabel
 1  26 Fe      # Species index, atomic number, species label
%endblock ChemicalSpeciesLabel

# Basis set definition
PAO.EnergyShift 200 meV
PAO.SplitNorm   0.15
PAO.BasisSize   DZP

# Lattice vectors
LatticeConstant  2.86 Ang
%block LatticeVectors
   -0.5000    0.5000    0.5000
    0.5000   -0.5000    0.5000
    0.5000    0.5000   -0.5000
%endblock LatticeVectors

#Atomic coordinates
AtomicCoordinatesFormat  ScaledCartesian

%block AtomicCoordinatesAndAtomicSpecies
    0.0000    0.0000    0.0000   1           
%endblock AtomicCoordinatesAndAtomicSpecies

# Real space grid 
MeshCutoff 125.0 Ry

# K points
%block kgrid.MonkhorstPack
  4    0    0    0.5
  0    4    0    0.5
  0    0    4    0.5
%endblock kgrid.MonkhorstPack

# Smearing
OccupationFunction    MP
OccupationMPOrder     2
ElectronicTemperature 300 K

# Convergence of

### From the fdf file ,it cleary see that siesta run without spin at default

In [10]:
!siesta < fe_bcc.fdf > fe_bcc.out

Job completed


In [11]:
!grep -i spin fe_bcc.out


# Spin options
# Spin <OPTION>
There are spin-orbit semi-local pseudopotentials available
redata: Spin configuration                          = none
redata: Number of spin components                   = 1
redata: Spin spiral                                 = F
redata: Write spin charge (when)                    = none
redata: Fix the spin of the system                  =   F


In [21]:
!grep "Total =" fe_bcc.out


siesta:         Total =    -775.730998


### Now creates a new folder EX2 and run the siesta again with spin

In [22]:
%cd ..

/home/l-rishiraj/siesta-docs/work-files/tutorials/basic/magnetism/01_Fe_bcc


In [13]:
!mkdir -p EX2

In [14]:
!cp fe_bcc.fdf Fe.psf EX2/

In [28]:
%cd EX2

/home/l-rishiraj/siesta-docs/work-files/tutorials/basic/magnetism/01_Fe_bcc/EX2


In [16]:
ls

fe_bcc.fdf  Fe.psf


In [29]:
!cat fe_bcc.fdf

#General system specifications
SystemName          Iron (bcc)
SystemLabel         fe_bcc
NumberOfAtoms       1
NumberOfSpecies     1

%block ChemicalSpeciesLabel
 1  26 Fe      # Species index, atomic number, species label
%endblock ChemicalSpeciesLabel

# Basis set definition
PAO.EnergyShift 200 meV
PAO.SplitNorm   0.15
PAO.BasisSize   DZP

# Lattice vectors
LatticeConstant  2.86 Ang
%block LatticeVectors
   -0.5000    0.5000    0.5000
    0.5000   -0.5000    0.5000
    0.5000    0.5000   -0.5000
%endblock LatticeVectors

#Atomic coordinates
AtomicCoordinatesFormat  ScaledCartesian

%block AtomicCoordinatesAndAtomicSpecies
    0.0000    0.0000    0.0000   1           
%endblock AtomicCoordinatesAndAtomicSpecies

# Real space grid 
MeshCutoff 125.0 Ry

# K points
%block kgrid.MonkhorstPack
  4    0    0    0.5
  0    4    0    0.5
  0    0    4    0.5
%endblock kgrid.MonkhorstPack

# Smearing
OccupationFunction    MP
OccupationMPOrder     2
ElectronicTemperature 300 K

# Convergence of

In [17]:
!siesta < fe_bcc.fdf > fe_bcc.out

Job completed


In [18]:
!grep -i spin fe_bcc.out

# Spin options
 Spin polarized
NOTE on units for spin output: 
{S} is the spin magnetic moment, in units of the Bohr magneton,
assuming a g-factor for the electron spin of exactly 2.
The spin magnetic moment is then numerically equal to the charge inbalance,
in units of electrons, between the spin-up and -down channels.
There are spin-orbit semi-local pseudopotentials available
redata: Spin configuration                          = collinear
redata: Number of spin components                   = 2
redata: Spin spiral                                 = F
redata: Write spin charge (when)                    = scf
redata: Antiferro initial spin density              =   F
redata: Fix the spin of the system                  =   F
mix.SCF: Spin-component mixing                   all
     spin moment: {S} , |S| = {        0.0        0.0    2.53151 }     2.53151
     spin moment: {S} , |S| = {        0.0        0.0    2.53118 }     2.53118
     spin moment: {S} , |S| = {        0.0        0.0    2

In [26]:
!grep "Total =" fe_bcc.out


siesta:         Total =    -776.180055


In [3]:
%cd ..

/home/l-rishiraj/siesta-docs/work-files/tutorials/basic/magnetism/01_Fe_bcc


---
### 1.How large is the spin magnetic moment per unit cell?

From the SCF output:

spin moment: {S} , |S| = { 0.0  0.0  2.18750 }  2.18750

The last (converged) value gives the total magnetic moment:

$[
M \approx 2.19 \, \mu_B
]$

Using

$[
M = N_\uparrow - N_\downarrow
]$

this corresponds to about **two unpaired electrons per unit cell**, which is typical for ferromagnetic iron.

---

## 2.How does the total energy of the two calculations compare?

Comparing the two calculations:

- Non-spin calculation → higher total energy i.e **-775.730998** 
- Spin-polarized calculation → lower total energy i.e **-776.180055** 

$[
E_{\text{spin}} < E_{\text{non-spin}}
]$

Thus, including spin lowers the total energy of the system.

---

## 3.What does the difference in total energy imply for the stability of the ferromagnetic phase (compared to the non-magnetic phase)?
In Density Functional Theory, the **ground state is the configuration with the lowest energy**.

Since:

$[
E_{\text{spin}} < E_{\text{non-spin}}
]$

the **ferromagnetic (spin-polarized) phase is more stable** than the non-magnetic phase.

This energy lowering originates from the **exchange interaction**, which favors parallel spin alignment.

---

## Final Results

- Magnetic moment ≈ **2.19 μB per unit cell**
- Spin-polarized energy < Non-spin energy
- **Ferromagnetic state is the stable ground state of Fe**





---
# EX3: Breaking the Symmetry in Spin-Polarized SIESTA Calculations

## Why Breaking the Spin Symmetry Matters

To simulate magnetic materials, it is not enough to simply ask for a spin-polarized calculation. The self-consistent field (SCF) procedure finds the ground state by iterating on an initial guess for the **density matrix**. If the **spin-up and spin-down channels are perfectly symmetric in the starting guess**, the SCF may stay symmetric and never develop spin polarization — even for a material that *should* exhibit a magnetic moment. 

In other words:

- **Breaking the symmetry** between spin-up and spin-down in the initial guess is necessary to obtain a **spin-polarized (magnetic) solution**. 
- If symmetry persists in the initial guess, the final solution may be **non-magnetic even though the material really is magnetic**. 

## How SIESTA Handles Spin Symmetry

SIESTA **will break the spin symmetry by default** when you request `Spin polarized`. It constructs an **initial density matrix with a spin imbalance**, typically filling partially occupied shells according to Hund’s rules, which provides an initial magnetic moment for spin-active atoms. 

However, we can *control* that initial spin configuration using the `DM.InitSpin` block in the input file:

    %block DM.InitSpin
    1 0.0
    %endblock DM.InitSpin



This tells SIESTA to start with **zero magnetic moment on atom 1**.

By doing this, I intentionally suppress any initial spin imbalance in the density matrix. Both spin-up and spin-down electrons start equally populated.

So at the beginning of the calculation, the system is completely non-magnetic.

---

## Why I Might Want to Control the Initial Spin

Manually specifying the initial spin is useful because it gives me control over how the calculation starts.

I can:

• Start from a completely non-magnetic state and check whether the system naturally develops magnetism during SCF  
• Force a particular magnetic ordering, such as ferromagnetic or antiferromagnetic  
• Test how different starting guesses affect the final converged solution  
• Avoid relying only on SIESTA’s automatic heuristics  

In short, it allows me to explore different magnetic configurations more reliably.

---



## In Simple Words

If both spins start equal, nothing changes.

If I introduce a small imbalance, the system can relax to the correct magnetic state.

So, breaking the symmetry in the beginning helps SIESTA find the true magnetic ground state.
    
    

In [1]:
!pwd

/home/l-rishiraj/siesta-docs/work-files/tutorials/basic/magnetism/01_Fe_bcc


In [2]:
!mkdir -p EX3

In [3]:
ls

EX1/  EX2/  EX3/  fe_bcc.fdf  Fe.psf  Untitled.ipynb


In [4]:
!cp fe_bcc.fdf  Fe.psf EX3/

In [5]:
%cd EX3

/home/l-rishiraj/siesta-docs/work-files/tutorials/basic/magnetism/01_Fe_bcc/EX3


/home/l-rishiraj/miniconda3/lib/python3.10/site-packages/IPython/core/magics/osm.py:417: UserWarning: This is now an optional IPython functionality, setting dhist requires you to install the `pickleshare` library.
  self.shell.db['dhist'] = compress_dhist(dhist)[-100:]


In [7]:
ls

fe_bcc.fdf  Fe.psf


In [10]:
!cat fe_bcc.fdf

#General system specifications
SystemName          Iron (bcc)
SystemLabel         fe_bcc
NumberOfAtoms       1
NumberOfSpecies     1

%block ChemicalSpeciesLabel
 1  26 Fe      # Species index, atomic number, species label
%endblock ChemicalSpeciesLabel

# Basis set definition
PAO.EnergyShift 200 meV
PAO.SplitNorm   0.15
PAO.BasisSize   DZP

# Lattice vectors
LatticeConstant  2.86 Ang
%block LatticeVectors
   -0.5000    0.5000    0.5000
    0.5000   -0.5000    0.5000
    0.5000    0.5000   -0.5000
%endblock LatticeVectors

#Atomic coordinates
AtomicCoordinatesFormat  ScaledCartesian

%block AtomicCoordinatesAndAtomicSpecies
    0.0000    0.0000    0.0000   1           
%endblock AtomicCoordinatesAndAtomicSpecies

# Real space grid 
MeshCutoff 125.0 Ry

# K points
%block kgrid.MonkhorstPack
  4    0    0    0.5
  0    4    0    0.5
  0    0    4    0.5
%endblock kgrid.MonkhorstPack

# Smearing
OccupationFunction    MP
OccupationMPOrder     2
ElectronicTemperature 300 K

# Convergence of

In [11]:
!siesta < fe_bcc.fdf > fe_bcc.out

Job completed


In [13]:
!grep "InitSpin" fe_bcc.out


%block DM.InitSpin
%endblock DM.InitSpin
